03_letters/sirigithub/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+// letters returns a map of rune literals and its frequency in the given input string s
+func letters(s string) map[rune]int {
+	charecterFreq := make(map[rune]int)
```

Comment:
A minor comment, but "character" is spelt incorrectly here. However I'd suggest not using the word "character" at all. Go consciously avoided using it - using the word "rune" instead - because character has a history of mixed meaning - is it one character in a string, or an 8-bit value? Rune is the former, but not the latter. Go uses `byte` for the latter.

I'd suggest just using `r` for the range variable: `for _, r := range s` reads quite naturally - `s` is a string, `r` is a rune and the scope is very small so no problem with the short variable names.

Using `runeFreq` for the map name would also make sense here. The `Freq` naming is good - it makes it quite clear what is going on.

Reply:
Great points ! I suppose in my head  I was still thinking character with runes.
Done.